SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -47,2469 +47,2615 @@
+   * \param[in] iPoint - Point index.
    * \return Value of the flow density.
    */
-  inline virtual su2double GetDensity(unsigned short val_iSpecies) {return 0; }
+  inline virtual su2double GetDensity(Idx_t iPoint, Idx_t val_iSpecies) const { return 0.0; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \param[in] val_Species - Index of species s.
    * \return Value of the mass fraction of species s.
    */
-  inline virtual su2double GetMassFraction(unsigned short val_Species) {return 0.0;}
+  inline virtual su2double GetMassFraction(Idx_t iPoint, Idx_t val_Species) const { return 0.0; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \return Value of the flow energy.
    */
-  inline virtual su2double GetEnergy(void) {return 0; }
+  inline virtual su2double GetEnergy(Idx_t iPoint) const { return 0.0; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \return Pointer to the force projection vector.
    */
-  inline virtual su2double *GetForceProj_Vector(void) {return NULL; }
+  inline virtual su2double *GetForceProj_Vector(Idx_t iPoint) { return nullptr; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \return Pointer to the objective function source.
    */
-  inline virtual su2double *GetObjFuncSource(void) {return NULL; }
+  inline virtual su2double *GetObjFuncSource(Idx_t iPoint) { return nullptr; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \return Pointer to the internal boundary vector.
    */
-  inline virtual su2double *GetIntBoundary_Jump(void) {return NULL; }
+  inline virtual su2double *GetIntBoundary_Jump(Idx_t iPoint) { return nullptr; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \return Value of the eddy viscosity.
    */
-  inline virtual su2double GetEddyViscosity(void) {return 0; }
+  inline virtual su2double GetEddyViscosity(Idx_t iPoint) const { return 0.0; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \return Value of the flow enthalpy.
    */
-  inline virtual su2double GetEnthalpy(void) {return 0; }
+  inline virtual su2double GetEnthalpy(Idx_t iPoint) const { return 0.0; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \return Value of the flow pressure.
    */
-  inline virtual su2double GetPressure(void) {return 0; }
+  inline virtual su2double GetPressure(Idx_t iPoint) const { return 0.0; }
 
   /*!
    * \brief A virtual member.
+   * \param[in] iPoint - Point index.
    * \param[in] val_vector - Direction of projection.
    * \return Value of the projected velocity.
    */
-  inline virtual su2double GetProjVel(su2double *val_vector) {return 0; }
+  inline virtual su2double GetProjVel(Idx_t iPoint, const su2double *val_vector) const { return 0.0; }
 
   /*!
    * \brief A virtual membe
```

Comment:
The gradients and limiters of secondaries are not used anywhere (but data is allocated for them). Is this legacy?

Reply:
I am fine with removing secondaries, they are not needed within the NICFD functionality.